drivers/scheduler/k8s/k8s.go

```diff
@@ -1150,51 +1149,53 @@ func (k *K8s) destroyCoreObject(spec interface{}, opts map[string]bool, app *spe
 
 }
 
-func (k *K8s) createAutopilotObject(apParams *scheduler.AutopilotParameters) (*apapi.AutopilotRule, error) {
-	obj := &apapi.AutopilotRule{}
-	apRuleParams := apParams.AutopilotRuleParameters
-	obj.Name = apParams.Name
-	obj.Namespace = apParams.Namespace
-	obj.Labels = map[string]string{"creator": "torpedo"}
-	obj.Spec.PollInterval = apParams.PollInterval
-	obj.Spec.ActionsCoolDownPeriod = apRuleParams.ActionsCoolDownPeriod
-	obj.Spec.Selector.LabelSelector.MatchLabels = apRuleParams.MatchLabels //map[string]string{"name": "pgbench-data"}
-	obj.Spec.NamespaceSelector.LabelSelector.MatchLabels = map[string]string{"creator": "torpedo"}
-	for _, ruleExpression := range apRuleParams.RuleConditionExpressions {
-		exprVolumeUsage := &apapi.LabelSelectorRequirement{
-			Key:      ruleExpression.Key,
-			Operator: apapi.LabelSelectorOperator(ruleExpression.Operator),
-			Values:   ruleExpression.Values,
-		}
-		obj.Spec.Conditions.Expressions = append(obj.Spec.Conditions.Expressions, exprVolumeUsage)
+func (k *K8s) addAppSpecEnvironmentVariables(containerEnvVars []v1.EnvVar, appEnvVars []scheduler.EnvVar) []v1.EnvVar {
+
+	var tmpEnvMapVars = make(map[string]string)
+	var envVars []v1.EnvVar
+	for _, cEnvVar := range containerEnvVars {
+		tmpEnvMapVars[cEnvVar.Name] = cEnvVar.Value
 	}
-	for _, ruleAction := range apRuleParams.RuleActions {
-		actions := &apapi.RuleAction{
-			Name:   ruleAction.Name,
-			Params: ruleAction.Params,
-		}
-		obj.Spec.Actions = append(obj.Spec.Actions, actions)
+	for _, appEnvVar := range appEnvVars {
+		tmpEnvMapVars[appEnvVar.Name] = appEnvVar.Value
+	}
+	for key, value := range tmpEnvMapVars {
+		envVars = append(envVars, v1.EnvVar{
+			Name:  key,
+			Value: value,
+		})
 	}
 
-	logrus.Infof("Using Autopilot Object: %+v\n", obj)
-
-	return obj, nil
+	return envVars
 }
 
-func (k *K8s) createAutopilotRule(autopilotRule *apapi.AutopilotRule) (*apapi.AutopilotRule, error) {
+func (k *K8s) createAutopilotRule(autopilotRule apapi.AutopilotRule) (*apapi.AutopilotRule, error) {
 	k8sOps := k8sops.Instance()
-	apRule, err := k8sOps.CreateAutopilotRule(autopilotRule)
-	if errors.IsAlreadyExists(err) {
-		if apRule, err = k8sOps.GetAutopilotRule(autopilotRule.Name); err == nil {
-			logrus.Infof("Found existing AutopilotRule: %v", autopilotRule.Name)
-			return apRule, nil
+
+	t := func() (interface{}, bool, error) {
+		autopilotRule.Labels = map[string]string{
+			"creator": "torpedo",
```

Comment:
`"creator": "torpedo"` is now being used at 3 places in the code base. Use const.